lib/asserts.nix

```diff
@@ -0,0 +1,44 @@
+{ lib }:
+
+rec {
+
+  /* Print a trace message if pred is false.
+     Intended to be used to augment asserts with helpful error messages.
+
+     Example:
+       assertMsg false "nope"
+       => false
+       stderr> trace: nope
+
+       assert (assertMsg ("foo" == "bar") "foo is not bar, silly"); ""
+       stderr> trace: foo is not bar, silly
+       stderr> assert failed at …
+
+     Type:
+       assertMsg :: Bool -> String -> Bool
+  */
+  # TODO(Profpatsch): add tests that check stderr
+  assertMsg = pred: msg:
+    if pred
+    then true
+    else builtins.trace msg false;
```

Comment:
Sounds like a good improvement to me.

Reply:
Wait, how would that look in practice? Would it still be used like

```
assert (assertMsg ("foo" == "bar") "foo is not bar, silly"); ""
```

Reply:
Sure, the difference is just that `assert` here essentially becomes a no-op if `pred` doesn't match, because `throw` already takes care of raising an error before the actual assertion fails.

Reply:
Then the API stays the same, I see. I think we should do that.